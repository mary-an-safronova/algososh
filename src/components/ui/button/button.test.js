import renderer from "react-test-renderer";
import { Button } from "./button";
import {render, screen, fireEvent} from "@testing-library/react";

describe('Button', () => {
    it('should renders correctly Button with text', () => {
        const tree = renderer.create(<Button text="testing text" />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Button without text', () => {
        const tree = renderer.create(<Button />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Button with disabled', () => {
        const tree = renderer.create(<Button disabled={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should renders correctly Button with loader', () => {
        const tree = renderer.create(<Button isLoader={true} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("should renders correctly Button event click", () => {
        const mockFn = jest.fn();
        render(<Button text="Button onClick" onClick={mockFn} />);
        const button = screen.getByText("Button onClick");
        fireEvent.click(button);
        expect(mockFn).toHaveBeenCalled();
    });
});
