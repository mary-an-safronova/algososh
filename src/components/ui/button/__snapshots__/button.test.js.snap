// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should renders correctly Button with disabled 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button should renders correctly Button with loader 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Button should renders correctly Button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    testing text
  </p>
</button>
`;

exports[`Button should renders correctly Button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Button с текстом 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Текст проверки
  </p>
</button>
`;
